### ISSUE TITLE
Ability to exclude a dataset from a recursive snapshot...

### ISSUE THREAD

**Issue by sean-** [zfsnap/zfsnap#68]
We have a zpool that has a ton of datasets that come and go, but we also have swap on the same zpool.  We want to use the recursive snapshot feature without snapping the swap dataset.  I thought this existed, but maybe not?

**Comment by otraupe** [comment]
That would be nice. I have a backup data set there which shouldn't be snapshotted because the pool would be depleted in no time.

**Comment by Alexqw** [comment]
@sean- @otraupe I agree, this would be a very handy feature. I currently just work around it by calling targets in one call.

However, while this would be nice, I don't see a clear path towards making this work when creating snapshots; destroying snapshots shouldn't be a problem.

How do you see the syntax working for something like this?

---Alex

**Comment by otraupe** [comment]
Alex, thanks for the reply!

Could you script-wise translate the recursive call to the zfs 
snapshotting feature into single non-recursive ones?

Ole


Am 17.11.2015 um 11:04 schrieb Alex Waite:

**Comment by bougui** [comment]
Hello all,

we would love to see and exclude list option, even if we know we can destroy those no require snapshot with a simple script, but having an option to exclude some dataset from zfssnap would be a very big plus for us 👍 .

An option could be something similar to rsync --exclude 

see rsync man page
            --exclude=PATTERN       exclude dataset matching PATTERN
            --exclude-from=FILE     read exclude patterns from FILE

This would be great, because we would be able to provision a single ascii file matching dataset that we don't want snapshot for.

Thanks

Guillaume.

**Comment by bill-mcgonigle** [comment]
the other script uses the `com.sun:auto-snapshot=false` property setting to mark excludes.  I was attempting to switch over to zfsnap but ran into this problem.
We want to be able to add new filesystems ad hoc and have them snapshotted without editing the crontab but still be able to mark a  few "temp" filesystems as having no snapshots.

**Comment by utrenkner** [comment]
I was just looking for such an option in zfsnap. I would fully support @bougui 's suggestion. That syntax (but without the equal sign) is used also by tar, tarsnap.

From [FreeBSD's tar(1)](https://www.freebsd.org/cgi/man.cgi?manpath=freebsd-release-ports&query=tar&sourceid=opensearch):
```
     --exclude pattern
	     Do	not process files or directories that match the	specified pat-
	     tern.  Note that exclusions take precedence over patterns or
	     filenames specified on the	command	line.

     [...]

     -X	filename, --exclude-from filename
	     Read a list of exclusion patterns from the	specified file.	 See
	     --exclude for more	information about the handling of exclusions.
```
Would be great to have this also in zfsnap!
